pkg/logging/http.go

```diff
@@ -57,9 +60,118 @@ func (m *HTTPServerMiddleware) HTTPMiddleware(name string, next http.Handler) ht
 }
 
 func NewHTTPServerMiddleware(logger log.Logger, opts ...Option) *HTTPServerMiddleware {
+
 	o := evaluateOpt(opts)
 	return &HTTPServerMiddleware{
 		logger: log.With(logger, "protocol", "http", "http.component", "server"),
 		opts:   o,
 	}
 }
+
+// Returns the logging ENUM based on logStart and logEnd values.
+func getHTTPLoggingOption(logStart bool, logEnd bool) (Decision, error) {
+	if !logStart && !logEnd {
+		return NoLogCall, nil
+	}
+
+	if !logStart && logEnd {
+		return LogFinishCall, nil
+	}
+
+	if logStart && logEnd {
+		return LogStartAndFinishCall, nil
+	}
+
+	return -1, fmt.Errorf("log start call is not supported.")
+}
+
+func getLevel(lvl string) level.Option {
+	switch lvl {
+	case "INFO":
+		return level.AllowInfo()
+	case "DEBUG":
+		return level.AllowDebug()
+	case "WARN":
+		return level.AllowWarn()
+	case "ERROR":
+		return level.AllowError()
+	default:
+		return level.AllowAll()
+	}
+}
+
+func NewHTTPLoggingOption(configYAML []byte) ([]Option, error) {
+
+	// Define a black config option.
+	logOpts := []Option{
+		WithDecider(func(_ string, err error) Decision {
+			return NoLogCall
+		}),
+	}
+
+	// If req logging is disabled.
+	if len(configYAML) == 0 {
+		return logOpts, nil
+	}
+
+	reqLogConfig, err := NewReqLogConfig(configYAML)
+	// If unmarshalling is an issue.
+	if err != nil {
+		return logOpts, err
+	}
+
+	globalLevel, globalStart, globalEnd, err := fillGlobalOptionConfig(reqLogConfig, false)
+
+	// If global options have invalid entries.
+	if err != nil {
+		return logOpts, err
+	}
+	// If the level entry does not matches our entries.
+	if err := validateLevel(globalLevel); err != nil {
+		// fmt.Printf("HTTP")
+		return logOpts, err
+	}
+
+	// If the combination is valid, use them, otherwise return error.
+	reqLogDecision, err := getHTTPLoggingOption(globalStart, globalEnd)
+	if err != nil {
+		return logOpts, err
+	}
+
+	logOpts = []Option{
+		WithFilter(func(logger log.Logger) log.Logger {
```

Comment:
This `Option` helps in applying filtered logging - https://pkg.go.dev/github.com/go-kit/kit/log/level#AllowAll

Same for gRPC Middleware doesn't exist, we can add this for gRPC Middleware